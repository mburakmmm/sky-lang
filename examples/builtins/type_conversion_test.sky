# Type conversion and numeric functions test

import _dummy

function main
  print("=== TYPE CONVERSIONS ===")
  
  # int() conversions
  print(int(3.14))
  print(int("42"))
  print(int("1010", 2))
  print(int("FF", 16))
  print(int(true))
  print(int(false))
  
  # float() conversions
  print(float(42))
  print(float("3.14"))
  print(float(true))
  
  # bool() conversions
  print(bool(1))
  print(bool(0))
  print(bool("hello"))
  print(bool(""))
  
  # str() conversions
  print(str(42))
  print(str(3.14))
  print(str(true))
  
  print("=== NUMERIC FUNCTIONS ===")
  
  # abs()
  print(abs(-10))
  print(abs(-3.14))
  print(abs(5))
  
  # min()
  print(min(5, 3, 8, 1))
  print(min(3.14, 2.5, 4.0))
  
  # max()
  print(max(5, 3, 8, 1))
  print(max(3.14, 2.5, 4.0))
  
  # round()
  print(round(3.14159, 2))
  print(round(2.7))
  
  # pow()
  print(pow(2, 3))
  print(pow(2, 10))
  
  # sqrt()
  print(sqrt(16))
  print(sqrt(2))
  
  # floor()
  print(floor(3.7))
  print(floor(3.2))
  
  # ceil()
  print(ceil(3.2))
  print(ceil(3.7))
  
  print("=== ALL TESTS PASSED ===")
end

