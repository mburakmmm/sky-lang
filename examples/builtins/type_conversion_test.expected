=== TYPE CONVERSIONS ===
3
42
10
255
1
0
42.000000
3.140000
1.000000
true
false
true
false
42
3.140000
true
=== NUMERIC FUNCTIONS ===
10
3.140000
5
1
2.500000
8
4.000000
3.140000
3.000000
8
1024
4.000000
1.414214
3
3
4
4
=== ALL TESTS PASSED ===

